tests/datatypes/test_bytes.py

```diff
@@ -63,31 +63,36 @@ def test_find(self):
 
     def test_isdigit(self):
         self.assertCodeExecution("""
-            for byt in [b'123', b'678', b'99999', b'abc', b'2s3x', 'a2b3', b'2S3X', 'A2b3', b'6', b'k', 'GHI', 'Jkl']:
+            for byt in [b'123', b'678', b'99999', b'abc', b'2s3x', 'a2b3', b'2S3X', 'A2b3', b'6',
+                    b'k', b'GHI', b'Jkl', b'!234', b'.', b'?/']:
```

Comment:
Sorry, I wasn't clear.
The test case that triggers the problem is punctuation + lowercase characters, like `b'.a.'` (and `b'.A.'` for isupper)

```
$ python
Python 3.4.4 (default, Nov 13 2016, 11:58:00) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> b'.a.'.islower()
True
>>> b'.A.'.isupper()
True
```